### ISSUE TITLE
Field parameter batch_first should have True as the default.

### ISSUE THREAD

**Issue by iantbutler01** [pytorch/text#1116]
## 🚀 Feature

**Motivation**

I'm new to Torch coming from Keras and TF predominantly and from a new to the library perspective, many layers and loss functions in Torch proper seem to expect batch_first to be True where as in Torchtext that is False and it creates a disconnect between the main library and this extension. Making True the default will align more with Torch and reduce friction when writing models using Torchtext as the data loader. 

**Pitch**

Switch the default arg for batch_first to True from False.

**Alternatives**

**Additional context**

This may be addressed in the major overhaul you guys are planning/working on but I'm not sure what you're changing in terms of the datasets and iterators so this may or may not be a relevant suggestion.

**Comment by zhangguanheng66** [comment]
The Field class will be retired soon as legacy code and we plan not to maintain it anymore. Thanks for the suggestions and we will follow the routine in the core library.

**Comment by DrMatters** [comment]
> The Field class will be retired soon as legacy code and we plan not to maintain it anymore. Thanks for the suggestions and we will follow the routine in the core library.

What's coming as a replacement for the Field class? It would be great to see examples of application of non-legacy code, because now it's unclear how to do so. There's also only legacy code usage example available on the main page of repository (https://github.com/pytorch/text#data).

**Comment by zhangguanheng66** [comment]
> > The Field class will be retired soon as legacy code and we plan not to maintain it anymore. Thanks for the suggestions and we will follow the routine in the core library.
> 
> What's coming as a replacement for the Field class? It would be great to see examples of application of non-legacy code. There's also only legacy code usage example available on the main page of repository (https://github.com/pytorch/text#data).

We have a few examples in https://github.com/pytorch/text/tree/master/examples/data_pipeline, that are using the prototype building blocks, instead of the old `Field`, `Iterator`, `Batch` class.
